src/Exception/AbstractBaseContainerException.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Dhii\Data\Container\Exception;
+
+use Exception as RootException;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\Data\Container\ContainerAwareTrait;
+use Dhii\Util\Normalization\NormalizeIntCapableTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+
+/**
+ * Common functionality for container exceptions.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractBaseContainerException extends RootException
+{
+    /* Adds string normalization capability.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeStringCapableTrait;
```

Comment:
The whole point of `AbstractBaseContainerException` is to re-use code. These traits would have to be used twice, once for each concrete. On the other hand, this functionality is common to the concretes.

When you made that point last time, I agreed. I still think your point is valid; I just don't think it is really that applicable in this case, because it decreases re-usability. But if you feel strongly about it, I will re-consider.

Reply:
Yes, I understand that this was done to avoid having to type the same list of traits in multiple concretes. I have nothing against _that_ in particular.

What urks me a bit is the fact that classes like these accomplish nothing functionality-wise, and are just a bundle of unused methods that do not use each other, unlike our abstracts and traits.

Up till now, my opinion has been "if this class doesn't need it, don't include it". However, seeing you doing this over and over has made me reconsider that. I can definitely see how useful a set of common traits can be. I've also encountered a similar dilemma just yesterday of whatever to group common traits into another single trait.

Perhaps we ought to come to a conclusion to this once and for all, document it in `dhii/dhii` and maintain consistency throughout our packages.